llvm/test/CodeGen/AMDGPU/call-args-inreg-no-sgpr-for-csrspill-xfail.ll

```diff
@@ -1,22 +1,452 @@
+; CHECK-NEXT:    s_mov_b64 s[10:11], s[34:35]
+; CHECK-NEXT:    s_mov_b32 s12, s83
+; CHECK-NEXT:    s_mov_b32 s13, s82
+; CHECK-NEXT:    s_mov_b32 s14, s81
+; CHECK-NEXT:    s_mov_b32 s15, s80
+; CHECK-NEXT:    s_mov_b32 s0, s71
+; CHECK-NEXT:    s_mov_b32 s1, s70
+; CHECK-NEXT:    s_mov_b32 s2, s69
+; CHECK-NEXT:    s_mov_b32 s3, s68
+; CHECK-NEXT:    s_mov_b32 s16, s67
+; CHECK-NEXT:    s_mov_b32 s17, s66
+; CHECK-NEXT:    s_mov_b32 s18, s65
+; CHECK-NEXT:    s_mov_b32 s19, s64
+; CHECK-NEXT:    s_mov_b32 s20, s55
+; CHECK-NEXT:    s_mov_b32 s21, s54
+; CHECK-NEXT:    s_mov_b32 s22, s53
+; CHECK-NEXT:    s_mov_b32 s23, s52
+; CHECK-NEXT:    s_mov_b32 s24, s51
+; CHECK-NEXT:    s_mov_b32 s25, s50
+; CHECK-NEXT:    s_swappc_b64 s[30:31], s[28:29]
+; CHECK-NEXT:    ; implicit-def: $vgpr0
+; CHECK-NEXT:    ; implicit-def: $vgpr1
+; CHECK-NEXT:    ; implicit-def: $vgpr31
+; CHECK-NEXT:    s_xor_b64 exec, exec, s[86:87]
+; CHECK-NEXT:    s_cbranch_execnz .LBB2_1
+; CHECK-NEXT:  ; %bb.2:
+; CHECK-NEXT:    s_mov_b64 exec, s[84:85]
+; CHECK-NEXT:    v_readlane_b32 s87, v40, 31
+; CHECK-NEXT:    v_readlane_b32 s86, v40, 30
+; CHECK-NEXT:    v_readlane_b32 s85, v40, 29
+; CHECK-NEXT:    v_readlane_b32 s84, v40, 28
+; CHECK-NEXT:    v_readlane_b32 s83, v40, 27
+; CHECK-NEXT:    v_readlane_b32 s82, v40, 26
+; CHECK-NEXT:    v_readlane_b32 s81, v40, 25
+; CHECK-NEXT:    v_readlane_b32 s80, v40, 24
+; CHECK-NEXT:    v_readlane_b32 s71, v40, 23
+; CHECK-NEXT:    v_readlane_b32 s70, v40, 22
+; CHECK-NEXT:    v_readlane_b32 s69, v40, 21
+; CHECK-NEXT:    v_readlane_b32 s68, v40, 20
+; CHECK-NEXT:    v_readlane_b32 s67, v40, 19
+; CHECK-NEXT:    v_readlane_b32 s66, v40, 18
+; CHECK-NEXT:    v_readlane_b32 s65, v40, 17
+; CHECK-NEXT:    v_readlane_b32 s64, v40, 16
+; CHECK-NEXT:    v_readlane_b32 s55, v40, 15
+; CHECK-NEXT:    v_readlane_b32 s54, v40, 14
+; CHECK-NEXT:    v_readlane_b32 s53, v40, 13
+; CHECK-NEXT:    v_readlane_b32 s52, v40, 12
+; CHECK-NEXT:    v_readlane_b32 s51, v40, 11
+; CHECK-NEXT:    v_readlane_b32 s50, v40, 10
+; CHECK-NEXT:    v_readlane_b32 s49, v40, 9
+; CHECK-NEXT:    v_readlane_b32 s48, v40, 8
+; CHECK-NEXT:    v_readlane_b32 s39, v40, 7
+; CHECK-NEXT:    v_readlane_b32 s38, v40, 6
+; CHECK-NEXT:    v_readlane_b32 s37, v40, 5
+; CHECK-NEXT:    v_readlane_b32 s36, v40, 4
+; CHECK-NEXT:    v_readlane_b32 s35, v40, 3
+; CHECK-NEXT:    v_readlane_b32 s34, v40, 2
+; CHECK-NEXT:    v_readlane_b32 s31, v40, 1
+; CHECK-NEXT:    v_readlane_b32 s30, v40, 0
+; CHECK-NEXT:    s_mov_b32 s32, s33
+; CHECK-NEXT:    v_readlane_b32 s4, v40, 32
+; CHECK-NEXT:    s_or_saveexec_b64 s[6:7], -1
+; CHECK-NEXT:    buffer_load_dword v40, off, s[0:3], s33 ; 4-byte Folded Reload
+; CHECK-NEXT:    s_mov_b64 exec, s[6:7]
+; CHECK-NEXT:    s_mov_b32 s33, s4
+; CHECK-NEXT:    s_waitcnt vmcnt(0)
+; CHECK-NEXT:    s_setpc_b64 s[30:31]
   call void @external_void_func_a15i32_inreg_i32_inreg([15 x i32] inreg %arg0, i32 inreg %arg1)
```

Comment:
Can we add a similar case using twice the same argument for the call ? Something like:

```asm
  call void @external_void_func_i32_inreg_i32_inreg(i32 inreg %arg0, i32 inreg %arg0)
```